src/server/transport.ts

```diff
@@ -0,0 +1,71 @@
+import { Transport, TransportOptions, NoticeTransportPayload } from "../core/types"
+import { URL } from "url";
+import http from "http";
+import https from "https";
+import { getStats } from "./util";
+import { sanitize } from "../core/util";
+
+export class ServerTransport implements Transport {
+    send(options: TransportOptions, payload?: NoticeTransportPayload | undefined): Promise<{ statusCode: number; body: string; }> {
+        const {protocol} = new URL(options.endpoint)
+        const transport = (protocol === "http:" ? http : https)
+
+        return new Promise((resolve, reject) => {
+            let promise: Promise<void>;
+
+            // this should not be here. it should be done before reaching the transport layer
+            // it could be inside a beforeNotifyHandler, but is not possible at the moment because those handlers are synchronous
```

Comment:
This is something we really should change in future. The fix might actually be as simple as invoking each handler with `await`.🤔

Reply:
True, I already suggested this but it may be trickier to push to our users. It will be a breaking change.

Reply:
Is it really? You can call synchronous functions with `await`.The main difference is that errors will now be promise rejections.

Oh, except you mean that the `notify` will now end up being promise-based if we do that (since you can only use `await` from an async function).

Reply:
> Oh, except you mean that the notify will now end up being promise-based if we do that (since you can only use await from an async function).

This 👍